test/metabase/models/permissions/parse_test.clj

```diff
@@ -25,8 +25,10 @@
       "/download/db/3/schema/PUBLIC/"                 {:db {3 {:download {:schemas {"PUBLIC" :full}}}}}
       "/download/limited/db/3/schema/PUBLIC/"         {:db {3 {:download {:schemas {"PUBLIC" :limited}}}}}
       "/download/db/3/schema/PUBLIC/table/4/"         {:db {3 {:download {:schemas {"PUBLIC" {4 :full}}}}}}
-      "/download/limited/db/3/schema/PUBLIC/table/4/" {:db {3 {:download {:schemas {"PUBLIC" {4 :limited}}}}}})))
-
+      "/download/limited/db/3/schema/PUBLIC/table/4/" {:db {3 {:download {:schemas {"PUBLIC" {4 :limited}}}}}}
+      "/data-model/db/3/"                             {:db {3 {:data-model :all}}}
+      "/data-model/db/3/schema/PUBLIC/"               {:db {3 {:data-model {:schemas {"PUBLIC" :all}}}}}
+      "/data-model/db/3/schema/PUBLIC/table/4/"       {:db {3 {:data-model {:schemas {"PUBLIC" {4 :all}}}}}})))
```

Comment:
here's an observation from the dark magic that you need for unsigned longs in java: try putting in "/data-model/db/18446744073709551616/" and you get a negative number in this parsed thing. this is because you need to enact dark magics upon it to get it to act right (specifically, you need to shove it to `toUnsignedString` instead of just printing it or using it like a normal long). I think the rest of the spec and type checking and stuff gets this one, but still pretty funny

Reply:
if it doesn't it rapidly stops being funny and ends up as an escalation of perms thing. but I think it does, so it remains funny

Reply:
hmm...when I try calling `permissions->graph` with that example path I get an error `String value 18446744073709551616 exceeds range of unsigned long.` Where are you getting a negative number?

I _think_ this shouldn't be an issue either way because we have schema validation that the ID is non-negative, and you'd hope DB IDs wouldn't get that high anyway.